enhancements/installer/custom-dns.md

```diff
@@ -0,0 +1,577 @@
+---
+title: aws-custom-dns
+authors:
+  - @sadasu
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - @patrickdillon, for Installer aspects
+  - @Miciah, for Ingress Controller aspects
+  - @cybertron, for on-prem networking aspects
+approvers:
+  - @patrickdillon
+  - @sdodson
+  - @dhellmann
+  - @zaneb
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - @JoelSpeed
+creation-date: 2022-10-25
+last-updated: 2022-10-25
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - "https://issues.redhat.com/browse/CORS-1874"
+---
+
+# Custom DNS for AWS
+
+## Summary
+
+This enhancement adds the ability to use a pre-configured custom DNS solution
+for API Server and Ingress resolution on the AWS platform during OpenShift
+IPI installation on that platform.
+
+## Motivation
+
+Some customers do not want to use the DNS solution provided by the underlying
+cloud due to regulatory[ITAR](https://www.gov-relations.com/) or operational
+contraints. It is important for OpenShift to provide a way for these customers
+to use their own preferred DNS solution while supporting their IPI deployments.
+
+This external Custom DNS solution would be responsible for API and Ingress Server
+resolution only. OpenShift would be responsible for creating and managing a DNS
+solution for API-Int Server resolution. The cloud Load Balancer is still the
+preferred solution for providing load balancing services for API, API-Int and Ingress
```

Comment:
A little confused about this. If we are providing api-int and the user is providing an IP address to use for it, then presumably we are running keepalived as in on-prem? Why would we also have an ELB for api-int?

Reply:
The requirement is for us to use the cloud LB. So, can we get away with not using keepalived ?
 
I think there is an open discussion about whether the APi-Int IPs have to be 1. provided by the user 2. derived from private-subnets (already being provided by the user) 3. same as API server IPs (this will not provide the benefit of a split horizon DNS)